File: src/HtmlStorageService.js
import React, { useState } from 'react';

const HtmlStorageService = () => {
  const [htmlContent, setHtmlContent] = useState('');
  const [savedPageId, setSavedPageId] = useState('');

  const savePage = () => {
    fetch('http://localhost:7000/save', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/x-www-form-urlencoded',
      },
      body: `htmlContent=${encodeURIComponent(htmlContent)}`,
    })
    .then(response => response.json())
    .then(data => {
      console.log('Página salva com sucesso. ID:', data.id);
      setSavedPageId(data.id);
    })
    .catch(error => {
      console.error('Erro ao salvar a página:', error);
    });
  };

  const loadPage = (pageId) => {
    fetch(`http://localhost:7000/page/${pageId}`)
    .then(response => {
      if (!response.ok) {
        throw new Error('Erro ao carregar a página.');
      }
      return response.text();
    })
    .then(htmlContent => {
      // Atualizar o estado com o conteúdo da página carregada
      setHtmlContent(htmlContent);
      console.log('Página carregada com sucesso.');
    })
    .catch(error => {
      console.error('Erro ao carregar a página:', error);
    });
  };

  return (
    <div>
      <h2>Armazenamento de HTML</h2>
      <label htmlFor="htmlContent">Conteúdo HTML:</label>
      <textarea
        id="htmlContent"
        rows="4"
        cols="50"
        value={htmlContent}
        onChange={(e) => setHtmlContent(e.target.value)}
      ></textarea>
      <br />
      <button onClick={savePage}>Salvar Página</button>
      <br />
      {savedPageId && (
        <div>
          <p>Página salva com sucesso. ID: {savedPageId}</p>
          <button onClick={() => loadPage(savedPageId)}>Carregar Página</button>
        </div>
      )}
    </div>
  );
};

export default HtmlStorageService;

File: flu/lib/html_storage_service.dart
import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;



class HtmlStorageServiceApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Html Storage Service',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: HtmlStorageServicePage(),
    );
  }
}

class HtmlStorageServicePage extends StatelessWidget {
  final TextEditingController htmlContentController = TextEditingController();
  String savedPageId = '';

  Future<void> savePage() async {
    try {
      var response = await http.post(
        Uri.parse('http://localhost:4000/save'),
        headers: {'Content-Type': 'application/x-www-form-urlencoded'},
        body: {'htmlContent': htmlContentController.text},
      );

      var data = response.body;
      savedPageId = data;
      print('Página salva com sucesso. ID: $savedPageId');
    } catch (error) {
      print('Erro ao salvar a página: $error');
    }
  }

  Future<void> loadPage(String pageId) async {
    try {
      var response = await http.get(Uri.parse('http://localhost:4000/page/$pageId'));
      if (response.statusCode == 200) {
        var loadedHtmlContent = response.body;
        htmlContentController.text = loadedHtmlContent;
        print('Página carregada com sucesso.');
      } else {
        throw Exception('Erro ao carregar a página.');
      }
    } catch (error) {
      print('Erro ao carregar a página: $error');
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Html Storage Service'),
      ),
      body: Padding(
        padding: const EdgeInsets.all(16.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Text('Armazenamento de HTML'),
            TextField(
              maxLines: 4,
              minLines: 1,
              controller: htmlContentController,
              decoration: InputDecoration(labelText: 'Conteúdo HTML:'),
            ),
            SizedBox(height: 16),
            ElevatedButton(
              onPressed: () => savePage(),
              child: Text('Salvar Página'),
            ),
            SizedBox(height: 16),
            if (savedPageId.isNotEmpty)
              Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text('Página salva com sucesso. ID: $savedPageId'),
                  ElevatedButton(
                    onPressed: () => loadPage(savedPageId),
                    child: Text('Carregar Página'),
                  ),
                ],
              ),
          ],
        ),
      ),
    );
  }
}
